### PR TITLE
ART-3057: find-bugs basis Brew event awareness

### PR DESCRIPTION
**Pull request by vfreex**
According to @jupierce :

```
Let:
Z be a non-closed BZ in a monitored component
S2 be the current state (as in the moment we are scanning) of Z
S1 be the state of the Z at the moment of the basis
A be the set of state changes Z after the basis
F be the sweep states (MODIFIED, ON_QA, VERIFIED)
Then Z is swept in if all the following are true:
S1 ∈ F
S2 ∈ F
A | ∄v : v <= S1
In prose: if a BZ seems to qualify for a sweep currently and at the basis event, then all state changes after the basis event must be to a greater than the state which qualified the BZ at the basis event.
```

Note:
1. Because this PR is implemented before adding assembly lense to Elliott ([ART-3045](https://issues.redhat.com/browse/ART-3045)), the basis event in this PR is taken from the command line argument like `find-builds` verb. When [ART-3045](https://issues.redhat.com/browse/ART-3045) is landed, we can extend to read releases.yml for basis event and additional bugs defined in `assembly.issues`.
2. Unit tests will be added later.

### CODE REVIEW

**Review by jupierce — CHANGES_REQUESTED** [review]
Elegant and concise. The change request is my fault for stating that we should use the basis event for the sweep. I think a minor change to variable naming will make this ready to merge.

File: elliottlib/cli/find_bugs_cli.py [hunk 1]
```diff
@@ -203,6 +211,67 @@ def find_bugs_cli(runtime, advisory, default_advisory_type, mode, check_builds,
         search_flag = 'blocker+' if mode == 'blocker' else None
         bugs = bzutil.search_for_bugs(bz_data, status, flag=search_flag, filter_out_security_bugs=not(cve_trackers),
                                       verbose=runtime.debug)
+
+        if brew_event:
+            brew_api = koji.ClientSession(runtime.group_config.urls.brewhub or constants.BREW_HUB)
+            basis_timestamp = brew_api.getEvent(brew_event)["ts"]
```
[the remark this answers is not in the document]
Comment (vfreex) [reply elliottlib/cli/find_bugs_cli.py:217]: > The basis is a moment in time where all image builds are complete. That could be several hours after upstream was rebased into distgit.
This is a good point. I didn't even notice that before. I need some time to think about that and follow up later.

[the remark this answers is not in the document]
Comment (vfreex) [reply elliottlib/cli/find_bugs_cli.py:217]: I spent some time on thinking if we can treat basis event as the time when we cut off the release. It is possible but requires a lot of changes in our existing implementation.
Continuing to treat the basis event as the moment when all builds are complete and making `find-bugs` determine the approximate rebase time is probably is best path to accomplish this.

I will make this happen in the next push.

**Comment by vfreex** [comment]
@jupierce I pushed a new commit to include your review suggestions and add unit tests.

This PR doesn't include calculating sweep cutoff timestamp from the basis event because we have changed the meaning of `brew_event` from basis brew event to sweep cutoff brew event. I also updated the help message to clarify the meaning of that command line argument. The calculation will go to the PR that honors assembly config.

**Review by jupierce — APPROVED** [review]

**Pull request merged**